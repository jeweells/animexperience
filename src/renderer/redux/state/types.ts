import { FStatus, Optional, TopView } from '@shared/types'
import {
  AnimeInfo,
  DeepAnimeIdSearchResult,
  EpisodeInfo,
  FollowedAnime,
  RecentAnimeInfo,
  RecommendationInfo,
  WatchHistoryItem
} from '@shared/types'
import { FollowedAnimeWStatus } from '../reducers/followedAnimes'
import { PreAllowWatch } from '../reducers/invokedLink'
import { RecentAnimeData } from '../../src/hooks/useRecentAnimes'
import { VideoOption } from '../../src/components/VideoPlayer'

export type AnimeSearchState = {
  // Boolean id
  searching?: string
  result?: Optional<DeepAnimeIdSearchResult>
  status: Partial<{
    result: FStatus
    moreResults: FStatus
  }>
}

export type FollowedAnimesState = {
  followedDict: Record<string, FollowedAnime>
  followed: FollowedAnimeWStatus[]
  status: Partial<{
    followed: FStatus
  }>
}

export type InvokedLinkState = {
  open: {
    watch?: string
  }
  preAllow: Partial<{
    watch: PreAllowWatch
  }>
  status: {
    watch?: FStatus
  }
}

export type NotificationsState = {
  message: string | null
  key: string
}

export type PeekState = {
  name?: string
  status: Partial<{
    info: FStatus
  }>
  info?: Optional<AnimeInfo>
  peeking?: string
}

export type PlayerState = {
  open?: string
  freezed?: boolean
}

export type OptionInfo = {
  name: string
  prefer?: boolean
  score: number
}
// Define a type for the slice state
export type PlayerOptionsState = {
  options?: OptionInfo[]
  history?: string[]
  preferred?: string[]
  status: Partial<{
    options: FStatus
  }>
}

export type RecommendationsState = {
  [name: string]: Partial<{
    status: FStatus
    recommendations: RecommendationInfo[]
  }>
}

export type TopviewState = {
  views: TopView[]
}

export type WatchState = {
  watching?: Optional<RecentAnimeData>
  availableVideos?: Optional<VideoOption[]>
  status: {
    availableVideos?: FStatus
    info?: FStatus
    watchEpisode?: FStatus
  }
  info?: Optional<AnimeInfo>
  showNextEpisodeButton?: boolean
  nextEpisodeTimeout: number
  autoFullScreen?: boolean
}

export type WatchedState = {
  episodes: {
    [name: string]: EpisodeInfo
  }
  recently?: RecentAnimeInfo[]
  status: {
    episodes?: FStatus
    recently?: FStatus
  }
}

export type WatchHistoryState = {
  sorted?: WatchHistoryItem[]
  status: {
    sorted?: FStatus
  }
}
