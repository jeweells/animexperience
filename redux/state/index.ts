import {
    AnimeSearchState,
    FollowedAnimesState,
    InvokedLinkState,
    NotificationsState,
    PeekState,
    PlayerOptionsState,
    PlayerState,
    RecommendationsState,
    TopviewState,
    WatchedState,
    WatchHistoryState,
    WatchState,
} from './types'

const animeSearch: AnimeSearchState = {
    status: {},
}

const followedAnimes: FollowedAnimesState = {
    followed: [],
    followedDict: {},
    status: {},
}

const invokedLink: InvokedLinkState = {
    open: {},
    preAllow: {},
    status: {},
}

const notifications: NotificationsState = {
    message: null,
    key: '_',
}

const peek: PeekState = {
    status: {},
}

const player: PlayerState = {}

const playerOptions: PlayerOptionsState = {
    status: {},
}

const recommendations: RecommendationsState = {}

const topView: TopviewState = {
    views: [],
}

const watch: WatchState = {
    status: {},
    nextEpisodeTimeout: -1,
}

const watched: WatchedState = {
    episodes: {},
    status: {},
}

const watchHistory: WatchHistoryState = {
    status: {},
}

export const initialState = {
    animeSearch,
    followedAnimes,
    invokedLink,
    notifications,
    peek,
    player,
    playerOptions,
    recommendations,
    topView,
    watch,
    watched,
    watchHistory,
}

export type RootState = typeof initialState
