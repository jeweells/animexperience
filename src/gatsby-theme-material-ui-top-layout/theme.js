import { createTheme } from '@material-ui/core/styles'

export const mainColor = 'rgb(105, 190, 213)'

const theme = createTheme({
    typography: {
        fontFamily: ['Quicksand', 'sans-serif'].join(','),
    },
    palette: {
        primary: {
            main: '#292d33',
            contrastText: '#e9ebf0',
        },
        secondary: {
            main: '#3c3f43',
        },
        text: {
            primary: '#e9ebf0',
            secondary: '#e9ebf0',
            disabled: '#ffffff82',
        },
    },
    overrides: {
        MuiButtonBase: {
            root: {
                borderRadius: undefined,
            },
        },
        MuiButton: {
            root: {
                color: '#e9ebf0',
                backgroundColor: '#292d33',
                marginBottom: 0,
                fontWeight: 'normal',
                textAlign: 'center',
                verticalAlign: 'middle',
                outline: '0 !important',
                whiteSpace: 'nowrap',
                border: 'none',
                userSelect: 'none',
                fontSize: 14,
                lineHeight: '1.42857143',
                borderRadius: 6,
                height: 36,
                '&:hover': {
                    backgroundColor: '#3c3f43',
                },
            },

            text: {
                color: '#e9ebf0',
                textTransform: 'none',
                padding: '8px 12px',
            },
        },
        MuiIconButton: {
            root: {
                flexShrink: 0,
                height: 36,
                minWidth: 36,
                color: '#e9ebf0',
                padding: '8px 12px',
                backgroundColor: '#292d33',
                marginBottom: 0,
                fontWeight: 'normal',
                textAlign: 'center',
                verticalAlign: 'middle',
                outline: '0 !important',
                whiteSpace: 'nowrap',
                border: 'none',
                userSelect: 'none',
                fontSize: 14,
                lineHeight: '1.42857143',
                borderRadius: 6,
                '&:hover': {
                    backgroundColor: '#3c3f43',
                },
            },
        },
        MuiCssBaseline: {
            '@global': {
                html: {
                    WebkitFontSmoothing: 'auto',
                },
                body: {
                    color: undefined,
                    margin: undefined,
                    fontSize: undefined,
                    fontFamily: undefined,
                    fontWeight: undefined,
                    backgroundColor: undefined,
                    lineHeight: undefined,
                },
            },
        },
        MuiSkeleton: {
            root: {
                backgroundColor: 'rgba(255, 255, 255, 0.11)',
            },
            pulse: {
                animation: 'MuiSkeleton-keyframes-pulse 3s ease-in 0.5s infinite',
            },
        },
    },
})

export default theme
