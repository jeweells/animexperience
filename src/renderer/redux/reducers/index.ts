export { animeSearch } from './animeSearch'
export { followedAnimes } from './followedAnimes'
export { invokedLink } from './invokedLink'
export { notifications } from './notifications'
export { peek } from './peek'
export { player } from './player'
export { playerOptions } from './playerOptions'
export { recommendations } from './recommendations'
export { topView } from './topView'
export { watch } from './watch'
export { watched } from './watched'
export { watchHistory } from './watchHistory'
