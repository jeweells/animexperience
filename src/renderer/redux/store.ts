import { configureStore } from '@reduxjs/toolkit'
import animeSearch from './reducers/animeSearch'
import followedAnimes from './reducers/followedAnimes'
import peek from './reducers/peek'
import player from './reducers/player'
import playerOptions from './reducers/playerOptions'
import recommendations from './reducers/recommendations'
import topView from './reducers/topView'
import watch from './reducers/watch'
import watched from './reducers/watched'
import watchHistory from './reducers/watchHistory'
import invokedLink from './reducers/invokedLink'
import notifications from './reducers/notifications'

export default configureStore({
  reducer: {
    watch,
    player,
    watched,
    recommendations,
    watchHistory,
    peek,
    topView,
    playerOptions,
    animeSearch,
    followedAnimes,
    invokedLink,
    notifications
  }
})
