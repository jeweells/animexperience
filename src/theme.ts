import { createTheme } from '@mui/material/styles'

const theme = createTheme({
    breakpoints: {
        values: {
            xs: 0,
            sm: 500,
            md: 800,
            lg: 1000,
            xl: 1200,
        },
    },
    typography: {
        fontFamily: ['Quicksand', 'sans-serif'].join(','),
    },
    palette: {
        mode: 'dark',
        primary: {
            main: '#292d33',
            contrastText: '#e9ebf0',
        },
        secondary: {
            main: '#3c3f43',
        },
        text: {
            primary: '#e9ebf0',
            secondary: '#e9ebf0',
            disabled: '#ffffff82',
        },
    },
    components: {
        MuiSvgIcon: {
            styleOverrides: {
                root: {
                    fontSize: 'inherit',
                },
            },
        },
        MuiButtonBase: {
            styleOverrides: {
                root: {
                    borderRadius: undefined,
                    textTransform: 'unset',
                },
            },
        },
        MuiButton: {
            styleOverrides: {
                root: {
                    color: '#e9ebf0',
                    backgroundColor: '#292d33',
                    marginBottom: 0,
                    fontWeight: 'normal',
                    textAlign: 'center',
                    verticalAlign: 'middle',
                    outline: '0 !important',
                    whiteSpace: 'nowrap',
                    border: 'none',
                    userSelect: 'none',
                    fontSize: 14,
                    lineHeight: '1.42857143',
                    textTransform: 'unset',
                    borderRadius: 6,
                    height: 36,
                    '&:hover': {
                        outline: 'none',
                        border: 'none',
                        backgroundColor: '#3c3f43',
                    },
                },
                text: {
                    color: '#e9ebf0',
                    textTransform: 'none',
                    padding: '8px 12px',
                },
            },
        },
        MuiIconButton: {
            styleOverrides: {
                root: {
                    flexShrink: 0,
                    height: 36,
                    minWidth: 36,
                    color: '#e9ebf0',
                    padding: '8px 6px',
                    backgroundColor: '#292d33',
                    marginBottom: 0,
                    fontWeight: 'normal',
                    textAlign: 'center',
                    verticalAlign: 'middle',
                    outline: '0 !important',
                    whiteSpace: 'nowrap',
                    border: 'none',
                    userSelect: 'none',
                    fontSize: '1.2rem',
                    lineHeight: '1.42857143',
                    borderRadius: 6,
                    '&:hover': {
                        backgroundColor: '#3c3f43',
                    },
                    '&.Mui-disabled': {
                        color: 'rgb(255 255 255 / 25%)',
                    },
                },
            },
        },
        MuiSkeleton: {
            defaultProps: {
                animation: 'pulse',
            },
            styleOverrides: {
                root: {
                    backgroundColor: 'rgba(255, 255, 255, 0.11)',
                },
                pulse: {
                    animationDuration: '1.5s 0.5s',
                },
            },
        },
        MuiCssBaseline: {
            styleOverrides: `
                * {
                    backface-visibility: hidden;
                    margin: 0;
                    padding: 0;
                    box-sizing: border-box;
                }
                html {
                    -webkit-font-smoothing: auto;
                }
                h6 {
                    margin: 0;
                }
                body {
                    background-color: #0F131A;
                }
                ::-webkit-scrollbar {
                    width: 10px;
                }
                ::-webkit-scrollbar-track {
                    background: transparent;
                }
                ::-webkit-scrollbar-thumb {
                    background: #25282e;
                },
                ::-webkit-scrollbar-thumb:hover {
                    background: #393a44;
                }
            `,
        },
    },
})

export default theme
