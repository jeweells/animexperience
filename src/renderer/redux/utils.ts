import { ThunkDispatch } from 'redux-thunk'
import { RootState } from './state'
import { AnyAction } from '@reduxjs/toolkit'
import { TypedUseSelectorHook, useDispatch, useSelector } from 'react-redux'
import { ForcedAny } from '@shared/types'

export type AppDispatch = ThunkDispatch<RootState, ForcedAny, AnyAction>

export const useAppSelector: TypedUseSelectorHook<RootState> = useSelector

export const useAppDispatch = useDispatch as () => AppDispatch
