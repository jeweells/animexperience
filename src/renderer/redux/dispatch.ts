import { AppDispatch } from '~/redux/utils'
import store from '~/redux/store'

export const dispatch: AppDispatch = (action) => store.dispatch(action)
