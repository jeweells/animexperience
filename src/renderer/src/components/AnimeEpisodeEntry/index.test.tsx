import { AnimeEpisodeEntry, AnimeEpisodeEntryProps } from './index'
import { ANIME_ENTRY_SELECTORS, ANIME_EPISODE_ENTRY } from '@selectors'
import { Provider } from 'react-redux'
import { RootState } from '~/redux/state'
import { DeepPartial } from 'redux'
import { ThemeProvider } from '@mui/material'
import theme from '../../theme'

describe('AnimeEpisodeEntry', () => {
  const returnTestId = ANIME_EPISODE_ENTRY.ANIME_INFO
  let initialState: DeepPartial<RootState>
  const props = {
    onClick: jest.fn(),
    onMouseOut: jest.fn(),
    onMouseOver: jest.fn(),
    anime: {
      name: 'Here goes the anime name',
      img: 'Here goes the anime image url',
      link: 'Here goes the anime link',
      episode: 69,
      date: "Here goes the date (I think it's not used anymore)"
    },
    management: {
      follow: false
    },
    isPopover: false,
    index: 0,
    visible: false,
    sliding: false
  }
  beforeEach(() => {
    initialState = {
      watched: {
        episodes: {
          69: {
            currentTime: 59,
            at: 101,
            duration: 600
          }
        }
      },
      followedAnimes: {
        followedDict: {
          [props.anime.name]: {
            name: props.anime.name,
            image: props.anime.img,
            link: 'Here goes anime link',
            lastCheckAt: 1,
            lastEpisodeWatched: 2,
            lastSuccessAt: 3,
            nextCheckAt: 4,
            nextEpisodeToWatch: 5
          }
        }
      }
    }
    props.onClick.mockClear()
    props.onMouseOut.mockClear()
    props.onMouseOver.mockClear()
  })

  const getComponent = (override: Partial<AnimeEpisodeEntryProps> = {}) => {
    return (
      <Provider store={mockStore(initialState)}>
        <ThemeProvider theme={theme}>
          <AnimeEpisodeEntry {...props} {...override} />
        </ThemeProvider>
      </Provider>
    )
  }

  it('renders default', async () => {
    const wrapper = render(getComponent())
    await waitFor(() => expect(wrapper.getByTestId(returnTestId)).toBeInTheDocument())
    // Since most tests depends on this component, make sure it's there
    expect(wrapper.getByTestId(ANIME_ENTRY_SELECTORS.WRAPPER)).toBeInTheDocument()
    expect(wrapper.baseElement).toMatchSnapshot()
  })

  it('renders when has follow management', async () => {
    const wrapper = render(getComponent({ management: { ...props.management, follow: true } }))
    await waitFor(() => expect(wrapper.getByTestId(returnTestId)).toBeInTheDocument())
    // Since most tests depends on this component, make sure it's there
    expect(wrapper.getByTestId(ANIME_ENTRY_SELECTORS.WRAPPER)).toBeInTheDocument()
    expect(wrapper.baseElement).toMatchSnapshot()
  })

  it('renders without anime', async () => {
    const wrapper = render(getComponent({ anime: null }))
    expect(wrapper.baseElement).toMatchSnapshot()
  })

  it('should pass anime on click', async () => {
    const wrapper = render(getComponent())
    await waitFor(() => expect(wrapper.getByTestId(returnTestId)).toBeInTheDocument())
    fireEvent.click(wrapper.getByTestId(ANIME_ENTRY_SELECTORS.HOVERABLE_AREA))
    expect(props.onClick).toHaveBeenCalledWith(props.anime)
  })
})
