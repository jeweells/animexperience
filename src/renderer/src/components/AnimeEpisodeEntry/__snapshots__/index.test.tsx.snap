// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AnimeEpisodeEntry renders default 1`] = `
.c3 {
  font-weight: 700;
  color: white;
  font-size: 1rem;
  line-height: 1;
  text-shadow: 0 0 2px black,0 0 6px black;
  margin-bottom: 4px;
  overflow: hidden;
  text-overflow: ellipsis;
  display: -webkit-box;
  -webkit-line-clamp: 1;
  -webkit-box-orient: vertical;
}

.c4 {
  font-weight: 300;
  color: white;
  font-size: 0.9rem;
  text-shadow: 0 0 2px black,0 0 4px black;
}

.c0 {
  position: relative;
  border-radius: 4px;
  overflow: hidden;
  --width: calc( ( 100vw - 3 * 16px - 2 * 72px ) / 4 );
  --height: calc(var(--width) * 0.64);
  width: var(--width);
  height: var(--height);
  -webkit-flex: 0 0 auto;
  -ms-flex: 0 0 auto;
  flex: 0 0 auto;
  cursor: pointer;
  -webkit-transition: all 300ms ease-in-out;
  transition: all 300ms ease-in-out;
}

.c0:hover {
  z-index: 10;
}

.c2 {
  position: absolute;
  top: 0;
  bottom: 0;
  left: 0;
  right: 0;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-box-pack: end;
  -webkit-justify-content: flex-end;
  -ms-flex-pack: end;
  justify-content: flex-end;
  background-image: linear-gradient( to top, rgba(0,0,0,0.7), rgba(0,0,0,0.5) 20%, rgba(0,0,0,0) 40% );
  padding: 16px;
}

.c1 {
  object-fit: cover;
  width: 100%;
  height: 100%;
}

@media (max-width:1000px) {
  .c0 {
    --width: calc( ( 100vw - 2 * 16px - 2 * 72px ) / 3 );
    --height: calc(var(--width) * 0.64);
  }
}

@media (max-width:800px) {
  .c0 {
    --width: calc( ( 100vw - 1 * 16px - 2 * 72px ) / 2 );
    --height: calc(var(--width) * 0.64);
    --height: calc(var(--width) * 0.84);
  }
}

@media (max-width:500px) {
  .c0 {
    --width: calc( ( 100vw - 0 * 16px - 2 * 72px ) / 1 );
    --height: calc(var(--width) * 0.64);
  }
}

<body>
  <div>
    <div
      data-testid="anime-entry-wrapper"
      style="opacity: 1; webkit-transition: opacity 1000ms cubic-bezier(0.4, 0, 0.2, 1) 0ms; transition: opacity 1000ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;"
    >
      <div
        class="c0"
        data-testid="anime-entry-background"
      >
        <img
          alt="Here goes the anime name"
          class="c1"
          src="Here goes the anime image url"
        />
        <div
          class="c2"
          data-testid="anime-episode-entry-anime-info"
        >
          <h3
            aria-label="Here goes the anime name"
            class="c3"
            data-mui-internal-clone-element="true"
          >
            Here goes the anime name
          </h3>
          <h6
            class="c4"
          >
            Episodio 69
          </h6>
        </div>
      </div>
    </div>
  </div>
</body>
`;

exports[`AnimeEpisodeEntry renders when has follow management 1`] = `
.c9 {
  font-weight: 700;
  color: white;
  font-size: 1rem;
  line-height: 1;
  text-shadow: 0 0 2px black,0 0 6px black;
  margin-bottom: 4px;
  overflow: hidden;
  text-overflow: ellipsis;
  display: -webkit-box;
  -webkit-line-clamp: 1;
  -webkit-box-orient: vertical;
}

.c10 {
  font-weight: 300;
  color: white;
  font-size: 0.9rem;
  text-shadow: 0 0 2px black,0 0 4px black;
}

.c7 {
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  width: 1em;
  height: 1em;
  display: inline-block;
  fill: currentColor;
  -webkit-flex-shrink: 0;
  -ms-flex-negative: 0;
  flex-shrink: 0;
  -webkit-transition: fill 200ms cubic-bezier(0.4,0,0.2,1) 0ms;
  transition: fill 200ms cubic-bezier(0.4,0,0.2,1) 0ms;
  font-size: 1.5rem;
  font-size: inherit;
}

.c8 {
  overflow: hidden;
  pointer-events: none;
  position: absolute;
  z-index: 0;
  top: 0;
  right: 0;
  bottom: 0;
  left: 0;
  border-radius: inherit;
}

.c5 {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  position: relative;
  box-sizing: border-box;
  -webkit-tap-highlight-color: transparent;
  background-color: transparent;
  outline: 0;
  border: 0;
  margin: 0;
  border-radius: 0;
  padding: 0;
  cursor: pointer;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  vertical-align: middle;
  -moz-appearance: none;
  -webkit-appearance: none;
  -webkit-text-decoration: none;
  text-decoration: none;
  color: inherit;
  text-transform: unset;
}

.c5::-moz-focus-inner {
  border-style: none;
}

.c5.Mui-disabled {
  pointer-events: none;
  cursor: default;
}

.c0 {
  position: relative;
  border-radius: 4px;
  overflow: hidden;
  --width: calc( ( 100vw - 3 * 16px - 2 * 72px ) / 4 );
  --height: calc(var(--width) * 0.64);
  width: var(--width);
  height: var(--height);
  -webkit-flex: 0 0 auto;
  -ms-flex: 0 0 auto;
  flex: 0 0 auto;
  cursor: pointer;
  -webkit-transition: all 300ms ease-in-out;
  transition: all 300ms ease-in-out;
}

.c0:hover {
  z-index: 10;
}

.c6 {
  -webkit-filter: drop-shadow(0px 0px 1px #000000aa);
  filter: drop-shadow(0px 0px 1px #000000aa);
  font-size: 24px;
}

.c4 {
  position: absolute;
  top: 0;
  left: 0;
  width: 100%;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: end;
  -webkit-justify-content: flex-end;
  -ms-flex-pack: end;
  justify-content: flex-end;
  padding: 8px;
  background-image: linear-gradient( to bottom, rgba(0,0,0,0.7), rgba(0,0,0,0.5) 20%, rgba(0,0,0,0) 90% );
  -webkit-backface-visibility: initial;
}

.c2 {
  position: absolute;
  top: 0;
  bottom: 0;
  left: 0;
  right: 0;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-box-pack: end;
  -webkit-justify-content: flex-end;
  -ms-flex-pack: end;
  justify-content: flex-end;
  background-image: linear-gradient( to top, rgba(0,0,0,0.7), rgba(0,0,0,0.5) 20%, rgba(0,0,0,0) 40% );
  padding: 16px;
}

.c2 .c3 {
  opacity: 0;
  -webkit-transition: opacity 300ms ease-in-out;
  transition: opacity 300ms ease-in-out;
}

.c2:hover .c3 {
  opacity: 1;
}

.c1 {
  object-fit: cover;
  width: 100%;
  height: 100%;
}

@media print {
  .c5 {
    color-adjust: exact;
  }
}

@media (max-width:1000px) {
  .c0 {
    --width: calc( ( 100vw - 2 * 16px - 2 * 72px ) / 3 );
    --height: calc(var(--width) * 0.64);
  }
}

@media (max-width:800px) {
  .c0 {
    --width: calc( ( 100vw - 1 * 16px - 2 * 72px ) / 2 );
    --height: calc(var(--width) * 0.64);
    --height: calc(var(--width) * 0.84);
  }
}

@media (max-width:500px) {
  .c0 {
    --width: calc( ( 100vw - 0 * 16px - 2 * 72px ) / 1 );
    --height: calc(var(--width) * 0.64);
  }
}

<body>
  <div>
    <div
      data-testid="anime-entry-wrapper"
      style="opacity: 1; webkit-transition: opacity 1000ms cubic-bezier(0.4, 0, 0.2, 1) 0ms; transition: opacity 1000ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;"
    >
      <div
        class="c0"
        data-testid="anime-entry-background"
      >
        <img
          alt="Here goes the anime name"
          class="c1"
          src="Here goes the anime image url"
        />
        <div
          class="c2"
          data-testid="anime-episode-entry-anime-info"
        >
          <div
            class="c3 c4"
          >
            <button
              aria-label="Seguir"
              class="c5 MuiButtonBase-root c6"
              data-mui-internal-clone-element="true"
              data-testid="manage-follow-button-button"
              tabindex="0"
              type="button"
            >
              <svg
                aria-hidden="true"
                class="c7 MuiSvgIcon-root MuiSvgIcon-fontSizeMedium"
                data-testid="FavoriteBorderRoundedIcon"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M19.66 3.99c-2.64-1.8-5.9-.96-7.66 1.1-1.76-2.06-5.02-2.91-7.66-1.1-1.4.96-2.28 2.58-2.34 4.29-.14 3.88 3.3 6.99 8.55 11.76l.1.09c.76.69 1.93.69 2.69-.01l.11-.1c5.25-4.76 8.68-7.87 8.55-11.75-.06-1.7-.94-3.32-2.34-4.28zM12.1 18.55l-.1.1-.1-.1C7.14 14.24 4 11.39 4 8.5 4 6.5 5.5 5 7.5 5c1.54 0 3.04.99 3.57 2.36h1.87C13.46 5.99 14.96 5 16.5 5c2 0 3.5 1.5 3.5 3.5 0 2.89-3.14 5.74-7.9 10.05z"
                />
              </svg>
              <span
                class="c8 MuiTouchRipple-root"
              />
            </button>
          </div>
          <h3
            aria-label="Here goes the anime name"
            class="c9"
            data-mui-internal-clone-element="true"
          >
            Here goes the anime name
          </h3>
          <h6
            class="c10"
          >
            Episodio 69
          </h6>
        </div>
      </div>
    </div>
  </div>
</body>
`;

exports[`AnimeEpisodeEntry renders without anime 1`] = `
<body>
  <div />
</body>
`;
